src/main/java/com/sap/oss/phosphor/fosstars/model/value/SemanticVersion.java

```diff
@@ -0,0 +1,81 @@
+package com.sap.oss.phosphor.fosstars.model.value;
+
+import java.util.Objects;
+import java.util.Optional;
+
+public class SemanticVersion {
+
+  private final int major;
+  private final int minor;
+  private final int micro;
+
+  private SemanticVersion(int major, int minor, int micro) {
```

Comment:
These fields seems to be not-negative integers. It may be good to add checks here but we can do it later.